apps/managedidentity/managedidentity_test.go

```diff
@@ -248,3 +376,369 @@ func TestCreatingIMDSClient(t *testing.T) {
+	}
+
+	// Test case: file does not exist
+	nonExistentFilePath := tmpFile.Name() + "_nonexistent"
+	if fileExists(nonExistentFilePath) {
+		t.Errorf("expected file not to exist, but it does")
+	}
+}
+
+func Test_handleAzureArcResponse(t *testing.T) {
+	testCaseFilePath, err := getMockFilePath()
+	if err != nil {
+		t.Fatalf("failed to get mock file path: %v", err)
+	}
+
+	testCases := []struct {
+		name           string
+		statusCode     int
+		headers        map[string]string
+		expectedError  string
+		platform       string
+		createMockFile bool
+		prepareMockEnv func(*testing.T)
+		cleanupMockEnv func()
+	}{
+		{
+			name:          "Not 401 error",
+			statusCode:    http.StatusOK,
+			headers:       map[string]string{},
+			expectedError: "managed identity error: 200",
+			platform:      runtime.GOOS,
+		},
+		{
+			name:          "No www-authenticate header",
+			statusCode:    http.StatusUnauthorized,
+			headers:       map[string]string{},
+			expectedError: "response has no www-authenticate header",
+			platform:      runtime.GOOS,
+		},
+		{
+			name:          "Basic realm= not found",
+			statusCode:    http.StatusUnauthorized,
+			headers:       map[string]string{wwwAuthenticateHeaderName: "Basic "},
+			expectedError: "basic realm= not found in the string",
+			platform:      runtime.GOOS,
+		},
+		{
+			name:          "Platform not supported",
+			statusCode:    http.StatusUnauthorized,
+			headers:       map[string]string{wwwAuthenticateHeaderName: "Basic realm=/path/to/secret.key"},
+			expectedError: "platform not supported",
+			platform:      "testPlatform",
+		},
+		{
+			name:           "Invalid file extension",
+			statusCode:     http.StatusUnauthorized,
+			headers:        map[string]string{wwwAuthenticateHeaderName: "Basic realm=/path/to/secret.txt"},
+			expectedError:  "invalid file extension",
+			platform:       runtime.GOOS,
+			createMockFile: true,
+		},
+		{
+			name:           "Invalid file path",
+			statusCode:     http.StatusUnauthorized,
+			headers:        map[string]string{wwwAuthenticateHeaderName: "Basic realm=/path/to/secret.key"},
+			expectedError:  "invalid file path",
+			platform:       runtime.GOOS,
+			createMockFile: true,
+		},
+		{
+			name:           "Unable to get file info",
+			statusCode:     http.StatusUnauthorized,
+			headers:        map[string]string{wwwAuthenticateHeaderName: "Basic realm=" + filepath.Join(testCaseFilePath, "2secret.key")},
+			expectedError:  "unable to get file info",
+			platform:       runtime.GOOS,
+			createMockFile: true,
+		},
+		{
+			name:           "Invalid secret file size",
+			statusCode:     http.StatusUnauthorized,
+			headers:        map[string]string{wwwAuthenticateHeaderName: "Basic realm=" + filepath.Join(testCaseFilePath, "secret.key")},
+			expectedError:  "invalid secret file size",
+			platform:       runtime.GOOS,
+			createMockFile: true,
+		},
+		// todo readd when we can figure out how to corrupt a file
```

Comment:
I don't think it is possible to mock the scenarios where the file is found at the file path expected by azure arc env.

Reply:
Was thinking the same but thought i would leave this here incase anybody had an idea

Reply:
Removed